components/eam/src/physics/cam/microp_aero.F90

```diff
@@ -616,31 +500,12 @@ subroutine microp_aero_run ( &
    do k = top_lev, pver
       do i = 1, ncol
 
-         select case (trim(eddy_scheme))
-         case ('diag_TKE', 'CLUBB_SGS')
-            wsub(i,k) = sqrt(0.5_r8*(tke(i,k) + tke(i,k+1))*(2._r8/3._r8))
-            wsub(i,k) = min(wsub(i,k),10._r8)
-            wsig(i,k) = max(0.001_r8, wsub(i,k))
-         case default 
-            ! get sub-grid vertical velocity from diff coef.
-            ! following morrison et al. 2005, JAS
-            ! assume mixing length of 30 m
-            dum = (kvh(i,k) + kvh(i,k+1))/2._r8/30._r8
-            ! use maximum sub-grid vertical vel of 10 m/s
-            dum = min(dum, 10._r8)
-            ! set wsub to value at current vertical level
-            wsub(i,k)  = dum
-         end select
-
-         if (eddy_scheme == 'CLUBB_SGS') then
-            wsubi(i,k) = max(0.2_r8, wsub(i,k))
-            wsubi(i,k) = min(wsubi(i,k), 10.0_r8)
-         else
-            wsubi(i,k) = max(0.001_r8, wsub(i,k))
-            if (.not. use_preexisting_ice) then
-               wsubi(i,k) = min(wsubi(i,k), 0.2_r8)
-            endif
-         endif
+         wsub(i,k) = sqrt(0.5_r8*(tke(i,k) + tke(i,k+1))*(2._r8/3._r8))
+         wsub(i,k) = min(wsub(i,k),10._r8)
+         wsig(i,k) = max(0.001_r8, wsub(i,k))
+        
+         wsubi(i,k) = max(0.2_r8, wsub(i,k))
+         wsubi(i,k) = min(wsubi(i,k), 10.0_r8)
```

Comment:
there is a function min_max_bound that can be applied. I got it from Balwinder:
```
   pure function min_max_bound(minlim, maxlim, input) result(bounded)

        real(r8), intent(in) :: minlim, maxlim
        real(r8), intent(in) :: input

        !return value
        real(r8) :: bounded

        bounded = max(min(maxlim, input), minlim)

   end function min_max_bound

```

Reply:
Now use common function min_max_bound